stfc/cloud-docs
# Add CI, treat warnings as errs and fix all errors

Following the shorter hackathon, I thought I'd look at the docs since that was the main topic. This PR:

- Adds CI to automatically build pull requests, any warnings will appear on the line in files changed so a reviewer can easily see if / where they are introducing a problem
- Fixes numerous short underlines, including one that broke the TOS link

See each commit message for more details

To test:
- Code review
- Check that it builds on CI and locally still


To see this in action:
https://github.com/DavidFair/cloud-docs/pull/1
(Since I cannot trigger a job without belonging to the repo)
